src/inmanta/ast/statements/generator.py

```diff
@@ -400,16 +400,23 @@ def execute(self, requires: Dict[object, object], resolver: Resolver, queue: Que
                 self, "attributes %s are part of an index and should be set in the constructor." % ",".join(missing_attrs)
             )
 
-        # the attributes
-        attributes = {k: v.execute(requires, resolver, queue) for (k, v) in self._direct_attributes.items()}
-        attributes.update(kwarg_attrs)
+        # schedule all direct attributes for direct execution
+        direct_attributes: Dict[str, object] = {
+            k: v.execute(requires, resolver, queue) for (k, v) in self._direct_attributes.items()
+        }
+        direct_attributes.update(kwarg_attrs)
+
+        # override defaults with kwargs
+        indirect_attributes: Dict[str, ExpressionStatement] = {
+            k: v for k, v in self._indirect_attributes.items() if k not in kwarg_attrs
```

Comment:
won't this allow things like
```
entity A:
  int a
end

v={"a":4}
A(a=3, **v)
```

Because a is indirect and kwarg?

Reply:
This is caught [above](https://github.com/inmanta/inmanta/pull/2005/files#diff-8b14cb4e126f41876f1b413531ac33f2L388-L391)
Should I clarify this in a comment?

Reply:
and ensure there is a testcase for that (it may be the case already)